### PR TITLE
fix: missing authority host param and resource management endpoint when init the client

### PR DESCRIPTION
**Pull request by locmai**
### Issue

Relates to https://github.com/tomkerkhove/promitor/issues/1646

### Description

This PR fixes 2 parts:
- First one is adding the AuthorityHost for the TokenCredentialOptions, since this value will be different for each Cloud
- Second one is similar to the first but with the ResourceGraphClient which will need the baseUri of the Managment Resource Endpoint

### CODE REVIEW

File: src/Promitor.Integrations.Azure/Authentication/AzureAuthenticationFactory.cs [hunk 1]
```diff
@@ -69,24 +69,26 @@ public static AzureAuthenticationInfo GetConfiguredAzureAuthentication(IConfigur
...
-                    tokenCredential = new ClientSecretCredential(tenantId, authenticationInfo.IdentityId, authenticationInfo.Secret);
+                    tokenCredential = new ClientSecretCredential(tenantId, authenticationInfo.IdentityId, authenticationInfo.Secret, tokenCredentialOptions);
                     break;
                 case AuthenticationMode.UserAssignedManagedIdentity:
-                    tokenCredential = new ManagedIdentityCredential(authenticationInfo.IdentityId);
+                    tokenCredential = new ManagedIdentityCredential(authenticationInfo.IdentityId, tokenCredentialOptions);
                     break;
                 case AuthenticationMode.SystemAssignedManagedIdentity:
-                    tokenCredential = new ManagedIdentityCredential();
+                    tokenCredential = new ManagedIdentityCredential( "",tokenCredentialOptions);
```
Comment (tomkerkhove) [inline src/Promitor.Integrations.Azure/Authentication/AzureAuthenticationFactory.cs:91]: Are we sure this is OK and won't blow up?

File: src/Promitor.Core/Extensions/AzureCloudExtensions.cs [hunk 2]
```diff
@@ -27,5 +28,22 @@ public static AzureEnvironment GetAzureEnvironment(this AzureCloud azureCloud)
                     throw new ArgumentOutOfRangeException(nameof(azureCloud), "No Azure environment is known for");
             }
         }
+
+        public static Uri GetAzureAuthorityHost(this AzureCloud azureCloud)
```
Comment (tomkerkhove) [inline src/Promitor.Core/Extensions/AzureCloudExtensions.cs:32]: Good thinking! Would you mind adding a unit test to ensure that it returns the correct values please?

Comment (tomkerkhove) [reply src/Promitor.Core/Extensions/AzureCloudExtensions.cs:32]: If not, happy to do a PR after this one - that's ok

File: src/Promitor.Agents.ResourceDiscovery/Graph/AzureResourceGraph.cs [hunk 3]
```diff
@@ -243,9 +243,13 @@ private async Task OpenConnectionAsync()
...
             var azureEnvironment = _resourceDeclarationMonitor.CurrentValue.AzureLandscape.Cloud.GetAzureEnvironment();
+            var azureAuthorityHost = _resourceDeclarationMonitor.CurrentValue.AzureLandscape.Cloud.GetAzureAuthorityHost();
 
-            var credentials = await AzureAuthenticationFactory.GetTokenCredentialsAsync(azureEnvironment.ManagementEndpoint, TenantId, _azureAuthenticationInfo);
-            var resourceGraphClient = new ResourceGraphClient(credentials);
+            var credentials = await AzureAuthenticationFactory.GetTokenCredentialsAsync(azureEnvironment.ManagementEndpoint, TenantId, _azureAuthenticationInfo, azureAuthorityHost);
+
+            var baseUri = new Uri(azureEnvironment.ResourceManagerEndpoint);
+
+            var resourceGraphClient = new ResourceGraphClient(baseUri ,credentials);
```
Comment (tomkerkhove) [inline src/Promitor.Agents.ResourceDiscovery/Graph/AzureResourceGraph.cs:252]: Nit
```suggestion
            var credentials = await AzureAuthenticationFactory.GetTokenCredentialsAsync(azureEnvironment.ManagementEndpoint, TenantId, _azureAuthenticationInfo, azureAuthorityHost);
            var resourceManagerBaseUri = new Uri(azureEnvironment.ResourceManagerEndpoint);

            var resourceGraphClient = new ResourceGraphClient(resourceManagerBaseUri, credentials);
```

**Comment by tomkerkhove** [comment]
It looks good, thank you for the PR! I'll wait with merging until you've been able to scrape it but happy to help where I can!

File: src/Promitor.Integrations.Azure/Authentication/AzureAuthenticationFactory.cs (hunk 1 above)
Comment (locmai) [reply src/Promitor.Integrations.Azure/Authentication/AzureAuthenticationFactory.cs:91]: Ahh I think I made a mistake here, should be

```
tokenCredential = new ManagedIdentityCredential(options:tokenCredentialOptions);
```
or 
```
tokenCredential = new ManagedIdentityCredential( default ,tokenCredentialOptions);
```

Comment (tomkerkhove) [reply src/Promitor.Integrations.Azure/Authentication/AzureAuthenticationFactory.cs:91]: Let's go for new ManagedIdentityCredential(options:tokenCredentialOptions); maybe?

File: src/Promitor.Core/Extensions/AzureCloudExtensions.cs (hunk 2 above)
Comment (locmai) [reply src/Promitor.Core/Extensions/AzureCloudExtensions.cs:32]: I updated the unit tests 😄

**Review by tomkerkhove — APPROVED** [review]

**Comment by locmai** [comment]
> It looks good, thank you for the PR! I'll wait with merging until you've been able to scrape it but happy to help where I can!

I am able to scrape the metrics and export them to my Prometheus after adding the azureMetadata.cloud in the configmap, the PR for the chart got merged too.

**Comment by tomkerkhove** [comment]
Awesome, thanks a ton @locmai!

**Pull request merged**